src/main/react/app/containers/App/excludedDate/excludedDate.js

```diff
@@ -0,0 +1,210 @@
+import React from "react";
+import DateInput from "./dateInput";
+import { DATE_TYPE, getDatePresets, PERIODS} from "../constants";
+import {
+  formatDate,
+  nextOccurrenceChineseLunar
+} from "../../../utils/date";
+import { getBrief } from "../../../utils";
+import moment from "moment";
+import {
+  RepeatCount,
+  RepeatInterval,
+  RepeatPeriod,
+  NameInput,
+  TimezoneInput,
+  PresetSelect,
+  RepeatCheckbox
+} from "./formItems";
+
+
+export default class ExcludeDate extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      selectedDateType: DATE_TYPE.TYPE_GREGORIAN,
+      selectedPreset: -1,
+      nextOccurrence: undefined,
+
+      name: "",
+      type: "",
+
+      utcOffset: moment().utcOffset(),
+      timezone:"",
+
+      startDate: {
+        dynamic: false,
+        date: new Date(),
+        dynamicDate: {
+          month: 1,
+          week: 1,
+          day: 1
+        }
+      },
+      endDate: {
+        dynamic: false, date: new Date(), dynamicDate: {
+          month: 1,
+          week: 1,
+          day: 1
+        }
+      },
+      timezones: [],
+      noEnd: true, //No end in repeat
+      repeat: true,
+      repeatCount: -1,
+      repeatInterval: 1,
+      repeatPeriod: PERIODS.Year
+    };
+  }
+
+  /*Apply the selected preset.*/
+  applyPreset = () => {
+    /*According to the selected date type, use different algorithms.*/
+    switch (this.state.selectedDateType) {
+      case DATE_TYPE.TYPE_GREGORIAN:
+        this.setState(this.getDatePresets()[this.state.selectedPreset]);
+        break;
+      case DATE_TYPE.TYPE_CHINESE_LUNAR:
+        let preset = this.getDatePresets()[this.state.selectedPreset];
+        this.setState(preset);
+
+        /*For types that are not gregorian, we calculate its occurrence
+        * using its corresponding algorithm.*/
+        this.setState({
+          nextOccurrence: nextOccurrenceChineseLunar(preset.params)
+        });
+    }
+  };
+
+  handleRepeatChange = (event) => {
+    this.setState({ repeat: event.target.checked });
+  };
+
+  handleRepeatPeriodChange = (e) => {
+    this.setState({ repeatPeriod: Number.parseInt(e.target.value) });
```

Comment:
Should we do something UI-wise the the value isn't parsable?

Reply:
That is, if  Number.parseInt can't parse the value, should we display an error message?